crates/tx5/src/url.rs

```diff
@@ -1,192 +1,323 @@
 use crate::*;
 
-fn parse_url<R: AsRef<str>>(r: R) -> Result<(String, Option<PubKey>)> {
-    let r = r.as_ref();
-    let url = ::url::Url::parse(r).map_err(Error::other)?;
-    match url.scheme() {
-        "ws" | "wss" => (),
-        scheme => {
-            return Err(Error::other(format!(
-                "invalid scheme, expected \"ws\" or \"wss\", got: {scheme:?}",
-            )));
-        }
-    }
-
-    if url.host_str().is_none() {
-        return Err(Error::other("InvalidHost"));
-    }
-
-    let pub_key = match url.path_segments() {
-        None => None,
-        Some(mut it) => match it.next() {
-            None => None,
-            Some(v) => {
-                if it.next().is_some() {
-                    return Err(Error::other("ExtraPathInfo"));
-                }
-                if v.is_empty() {
-                    None
-                } else {
-                    use base64::Engine;
-                    let dec = base64::engine::general_purpose::URL_SAFE_NO_PAD
-                        .decode(v)
-                        .map_err(Error::other)?;
-
-                    if dec.len() != 32 {
-                        return Err(Error::other("InvalidPubKey"));
-                    }
-
-                    Some(PubKey(Arc::new(dec.try_into().unwrap())))
-                }
-            }
-        },
-    };
-
-    Ok((r.to_string(), pub_key))
+struct UrlCanon {
```

Comment:
Internal struct so doc comment optional but I think it might be useful to have a brief description here?

Reply:
```suggestion
/// SigUrl and PeerUrl share a lot of common parsing code.
/// This internal struct ensures the base urls are "canonical",
/// always explicitly listing even well-known ports 80 and 443.
struct UrlCanon {
```